src/storage/query/QueryEdgePropsProcessor.cpp

```diff
@@ -27,17 +27,30 @@ kvstore::ResultCode QueryEdgePropsProcessor::collectEdgesProps(
         return ret;
     }
 
+    auto schema = this->schemaMan_->getEdgeSchema(spaceId_, std::abs(edgeKey.edge_type));
+    auto retTTLOpt = getEdgeTTLInfo(edgeKey.edge_type);
     // Only use the latest version.
     if (iter && iter->valid()) {
         RowWriter writer(rsWriter.schema());
         PropsCollector collector(&writer);
         auto reader = RowReader::getEdgePropReader(schemaMan_,
                                                    iter->val(),
                                                    spaceId_,
-                                                   edgeKey.edge_type);
+                                                   std::abs(edgeKey.edge_type));
+
+        // Check if ttl data expired
+        if (retTTLOpt.has_value()) {
+            auto ttlValue = retTTLOpt.value();
+            if (checkDataExpiredForTTL(schema.get(),
+                                       reader.get(),
+                                       ttlValue.first,
+                                       ttlValue.second)) {
+                iter->next();
```

Comment:
why next when expired.

Reply:
Because, line 54  `iter->next();`   I don’t think it ’s useful. I just wrote it for consistency.

Reply:
I removed them all